File: DevExpress-Demo/CS/MVVMDemo.Wpf/Modules/Services/DispatcherServiceViewModel.cs
using DevExpress.Mvvm.POCO;
using DevExpress.Mvvm;
using System;
using System.Threading;
using System.Threading.Tasks;
using System.Collections.ObjectModel;

namespace MVVMDemo.Services {
    public class DispatcherServiceViewModel {
        public ObservableCollection<string> Items { get; set; }

        public Task Generate() {
            return Task.Factory.StartNew(GenerateCore);
        }
        void GenerateCore() {
            IDispatcherService service = this.GetRequiredService<IDispatcherService>();
            service.BeginInvoke(() => Items.Clear());
            for(int i = 0; i <= 10; i++) {
                string item = "Item " + i;
                service.BeginInvoke(() => Items.Add(item));
                Thread.Sleep(TimeSpan.FromMilliseconds(100));
            }
        }

        protected DispatcherServiceViewModel() {
            Items = new ObservableCollection<string>();
        }
    }
}

File: DevExpress-Demo/VB/MVVMDemo.Wpf/Modules/Services/DispatcherServiceViewModel.vb
Imports DevExpress.Mvvm.POCO
Imports DevExpress.Mvvm
Imports System
Imports System.Threading
Imports System.Threading.Tasks
Imports System.Collections.ObjectModel

Namespace MVVMDemo.Services

    Public Class DispatcherServiceViewModel

        Public Property Items As ObservableCollection(Of String)

        Public Function Generate() As Task
            Return Task.Factory.StartNew(New Action(AddressOf GenerateCore))
        End Function

        Private Sub GenerateCore()
            Dim service As IDispatcherService = GetRequiredService(Of IDispatcherService)()
            service.BeginInvoke(Sub() Items.Clear())
            For i As Integer = 0 To 10
                Dim item As String = "Item " & i
                service.BeginInvoke(Sub() Items.Add(item))
                Thread.Sleep(TimeSpan.FromMilliseconds(100))
            Next
        End Sub

        Protected Sub New()
            Items = New ObservableCollection(Of String)()
        End Sub
    End Class
End Namespace
